### PR TITLE
390 change content in check tool

### PR DESCRIPTION
**Pull request by GeorgeGoodall-GovUk**
## What type of PR is this? (check all applicable)

- Refactor
- Feature

## Description
Content changes made to check tool following the design as seen in the [ticket](https://github.com/orgs/digital-land/projects/7/views/15?pane=issue&itemId=78872607&issue=digital-land%7Csubmit%7C390)

It also makes it so the dataset display widget doesn't need info from the deep link to work, but instead can use the selected form fields form earlier steps of the check tool

## Related Tickets & Documents
- Ticket Link: #571 

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, a note
on the devices and browsers this has been tested on, as well as any relevant
images for UI changes._

## Added/updated tests?
_We encourage you to keep the code coverage percentage at 80% and above._

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## Release Notes

- **New Features**
  - Introduced dynamic status content rendering with a new macro for improved user experience.
  - Enhanced the file upload component with clearer interaction options during errors.

- **Improvements**
  - Centralised text management for status messages and button texts for better modularity.
  - Updated back link text to be dynamic, improving navigation context.
  - Revised logic for rendering dataset banners to use dataset names instead of deep links.
  - Enhanced the confirmation page with updated instructions for publishing data, ensuring clarity for users.
  - Improved accessibility with new aria labels for buttons.
  - Simplified dataset descriptions across various components for better clarity.

- **Bug Fixes**
  - Streamlined script loading to ensure proper functionality of JavaScript components.

### CODE REVIEW

File: src/content/statusPage.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export const headingTexts = {
+  checking: 'Checking your data',
+  checked: 'Data Checked'
+}
+
+export const messageTexts = {
+  checking: 'Do not close this page',
+  checked: 'You can continue'
+}
```
[the remark this answers is not in the document]
Comment (GeorgeGoodall-GovUk) [reply src/content/statusPage.js:9]: this was a design decision

**Comment by DilwoarH** [comment]
@GeorgeGoodall-GovUk  - do you know what's happening with the playwright tests? why do they take so long to complete and why are they suddenly failing?

File: index.js [hunk 2]
```diff
@@ -13,9 +13,11 @@ import { setupSession } from './src/serverSetup/session.js'
 import { setupNunjucks } from './src/serverSetup/nunjucks.js'
 import { setupSentry } from './src/serverSetup/sentry.js'
 import { getDatasetSlugNameMapping } from './src/utils/datasetteQueries/getDatasetSlugNameMapping.js'
+import { initDatasetSlugToReadableNameFilter } from './src/utils/datasetSlugToReadableName.js'
 
 dotenv.config()
 
+await initDatasetSlugToReadableNameFilter()
```
[the remark this answers is not in the document]
Comment (GeorgeGoodall-GovUk) [reply index.js:20]: already handled in initDatasetSluigToReadableNameFilter()

File: test/unit/lpaDetailsController.test.js [hunk 3]
```diff
@@ -16,6 +17,7 @@ describe('lpaDetailsController', async () => {
     controller = new LpaDetailsController.default({
       route: '/lpa-details'
     })
+    await initDatasetSlugToReadableNameFilter()
```
[the remark this answers is not in the document]
Comment (GeorgeGoodall-GovUk) [reply test/unit/lpaDetailsController.test.js:20]: already handled in function

File: src/content/statusPage.js
Comment (rosado) [inline src/content/statusPage.js:1]: Probably makes sense for @alextea to review the capitalisation of these messages?

**Review by rosado — APPROVED** [review]

**Pull request merged**